app/models/pipeline_run.rb

```diff
@@ -295,21 +295,24 @@ def failed?
   end
 
   def create_output_states
-    # First, determine which outputs we need:
-    target_outputs = %w[input_validations ercc_counts taxon_counts contig_counts taxon_byteranges amr_counts]
-
-    # Then, generate output_states
-    output_state_entries = []
-    target_outputs.each do |output|
-      output_state_entries << OutputState.new(
-        output: output,
-        state: STATUS_UNKNOWN
-      )
-    end
-    self.output_states = output_state_entries
+    # If we create a sample with results, we should not trigger a new run
+    unless results_finalized?
```

Comment:
I feel like this is the wrong place to make a change. It makes the code really confusing ("how could `results_finalized` be true if the pipeline_run is just getting created?"). Can you not move the logic for skipping the callbacks to your rake task instead? https://stackoverflow.com/questions/632742/how-can-i-avoid-running-activerecord-callbacks

Reply:
Thats a good suggestion for this particular case. I also think we should avoid callbacks

Reply:
> Fun fact: copying the S3 files is not enough to make results_folder work, because
> 
> https://github.com/chanzuckerberg/idseq-web/blob/a5456d1b63389783aa48d050e343b0bbe514af73/app/models/pipeline_run.rb#L1353
> 
> uses the dag json as it was originally specified (i.e. with the old project id)

Should this be dynamic from https://github.com/chanzuckerberg/idseq-web/blob/master/app/models/sample.rb#L75 ?

Reply:
@cdebourcy Suggestion does not work since it is an old method.
There are suggestions that involve monkey patching `ActiveRecord::skip_callbacks` which seems bad practice.